### PR TITLE
Add now playing state channels

### PR DESCRIPTION
**Pull request by teancom**
## Is there anything about the code changes here that should be highlighted?

Publish track metadata, transport anchors, and queue modes as separate replayable StateFlows from MainDataSource. Transport change detection projects the previous anchor before applying the two-second tolerance, resets on track identity changes, and collapses frozen playback to rate zero. Add focused tests for metadata mapping, null state, transport deduplication, freeze handling, and shuffle/repeat availability gates.

## Are there any additional changes not related to the issue above?

This one is pure scaffolding (and tests for the scaffolding) for the next couple of PRs. Nothing actually reads / uses this code, yet.

## Before submitting this PR, make sure you have:
- [X] Given this PR a readable name that can be used in the app's changelog
- [X] Made sure checks pass by running `./gradlew detektAll :composeApp:testAndroidHostTest :androidApp:testDebug`

### CODE REVIEW

File: composeApp/src/commonMain/kotlin/io/music_assistant/client/data/NowPlayingChannels.kt [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        old: NowPlayingTransport?,
+        newMediaItemId: String?,
+        new: NowPlayingTransport?,
+    ): Boolean {
+        if (oldMediaItemId != newMediaItemId) return false
+        if (old == null || new == null) return old == new
+        if (old.isPlaying != new.isPlaying || old.rate != new.rate) return false
+
+        return when {
+            old.elapsedSec == null && new.elapsedSec == null -> true
```
Comment (formatBCE) [inline composeApp/src/commonMain/kotlin/io/music_assistant/client/data/NowPlayingChannels.kt]: You can use same thing for uniformity:
`if (old.elapsedSec == null || new.elapsedSec == null) return old.elapsedSec == new.elapsedSec`
and then 
```
               val projectedOld = old.elapsedSec +
                    (new.anchorMs - old.anchorMs) / 1000.0 * old.rate
                abs(projectedOld - new.elapsedSec) < ELAPSED_ANCHOR_EPSILON_S
```
It flattens that `when`

File: composeApp/src/commonMain/kotlin/io/music_assistant/client/data/NowPlayingChannels.kt [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        }
+    }
+}
+
+/** Maps local-player state to the metadata channel. */
+internal fun buildNowPlayingTrack(playerData: PlayerData?): NowPlayingTrack? =
+    buildNowPlayingTrackFromItem(playerData?.queueInfo?.currentItem?.track)
+
+/** Maps one playable item without requiring a full player or queue fixture. */
+internal fun buildNowPlayingTrackFromItem(item: PlayableItem?): NowPlayingTrack? =
```
Comment (formatBCE) [inline composeApp/src/commonMain/kotlin/io/music_assistant/client/data/NowPlayingChannels.kt]: This method is used only in one place (except tests), and can be inlined.

**Review by formatBCE — COMMENTED** [review]
Comments. Nothing bad, just some nit picks.

**Review by formatBCE — APPROVED** [review]
Feel free to fix in next PR.

**Comment by teancom** [comment]
@formatBCE I'm heading to bed so am not starting on the next phase tonight. But had enough brain power left to cleane up those two nits. 😄

**Review by formatBCE — APPROVED** [review]
